core/evm.go

```diff
@@ -61,6 +61,8 @@ func NewEVMBlockContext(header *types.Header, chain ChainContext, author *common
 		Difficulty:  new(big.Int).Set(header.Difficulty),
 		BaseFee:     baseFee,
 		GasLimit:    header.GasLimit,
+		Random:      header.MixDigest,
+		IsPostMerge: header.Difficulty.Cmp(common.Big0) == 0,
```

Comment:
Do we need this as a field though? We do pass in the difficult anyway, couldn't we just compare in there? Or look at the `random` field and see if it's set or not?